app/controllers/application_controller.rb

```diff
@@ -317,6 +317,8 @@ def configure_permitted_parameters
     # Otherwise owners should go to their dashboards
     # And everyone else should go to user dashboard/watchlist
   def after_sign_in_path_for(resource)
+    update_search_attempt_user(current_user)
```

Comment:
I'd expect `after_sign_in_path_for` to be a read-only method, so adding this call to update the search attempt user creates an unexpected side-effect.  Is there a way to move the call to whatever calls `after_sign_in_path` or to possibly use a Devise-specific call-back to execute `update_search_attempt_user`?

Reply:
Good idea, just moved it to a Devise/Warden [`after_set_user` callback](https://github.com/wardencommunity/warden/wiki/callbacks#after_set_user).